core/cloudflow-akka-tests/src/test/scala/cloudflow/akkastream/scaladsl/AkkaStreamletConsumerGroupSpec.scala

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (C) 2016-2020 Lightbend Inc. <https://www.lightbend.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cloudflow.akkastream.util.scaladsl
+
+import scala.concurrent._
+import scala.concurrent.duration._
+
+import akka.NotUsed
+import akka.actor.ActorSystem
+import akka.stream.scaladsl._
+
+import com.typesafe.config._
+import cloudflow.akkastream._
+import cloudflow.akkastream.scaladsl._
+import cloudflow.akkastream.testdata._
+import cloudflow.streamlets._
+import cloudflow.streamlets.avro._
+
+import net.manub.embeddedkafka._
+import org.scalatest.time._
+
+object AkkaStreamletConsumerGroupSpec {
+  val kafkaPort = 1234
+  val zkPort    = 5678
+  val config    = ConfigFactory.parseString(s"""
+      akka {
+        stdout-loglevel = "OFF"
+        loglevel = "OFF"
+      }
+      cloudflow.kafka.bootstrap-servers = "localhost:$kafkaPort"
+      """)
+}
+
+import AkkaStreamletConsumerGroupSpec._
+
+class AkkaStreamletConsumerGroupSpec extends EmbeddedKafkaSpec(kafkaPort, zkPort, ActorSystem("test", config)) {
+
+  implicit override val patienceConfig = PatienceConfig(timeout = Span(10, Seconds), interval = Span(20, Millis))
+
+  override def createKafkaConfig: EmbeddedKafkaConfig =
+    EmbeddedKafkaConfig(kafkaPort,
+                        zooKeeperPort,
+                        Map(
+                          "broker.id"                        -> "1",
+                          "num.partitions"                   -> "53",
+                          "offsets.topic.replication.factor" -> "1",
+                          "offsets.topic.num.partitions"     -> "3"
+                        ))
+
+  "Akka streamlet instances" should {
+    "consume from an outlet as a group for the same streamlet" in {
```

Comment:
```suggestion
    "consume from an outlet as a group per unique streamlet name streamlet" in {
```

Reply:
"per unique streamlet name streamlet" reads weird..
I'll do "consume from an outlet as a group per streamlet reference"

Reply:
I would add the word unique.

I think "streamlet reference" is kind of an internal implementation name that perhaps not everybody understand right away but I can live with it 😉